pkg/container-collection/options.go

```diff
@@ -456,50 +486,32 @@ func WithKubernetesEnrichment(nodeName string, kubeconfig *rest.Config) Containe
 
 		// Future containers
 		cc.containerEnrichers = append(cc.containerEnrichers, func(container *Container) bool {
-			if container.K8s.PodName != "" {
+			// Skip enriching if all k8s fields are already known.
+			// This is an optimization and to make sure to avoid erasing the fields in case of error.
+			if runtimeclient.IsEnrichedWithK8sMetadata(container.K8s.BasicK8sMetadata) &&
+				container.K8s.PodLabels != nil {
```

Comment:
IsEnrichedWithK8sMetadata() already checks PodLabels

Reply:
In this PR, this is not the case.
Only in the other PR at this commit: https://github.com/inspektor-gadget/inspektor-gadget/pull/2398/commits/1b3feb537f7781133edd2ebb8b90dcd5df207fee#diff-8e49337a1417a39a62ebf427e84fb2faa566c96e37ae627ac484c05dbe65c483R149